### PR TITLE
fix: limit action name chars

### PR DESCRIPTION
**Pull request by angrybayblade**
# 🔍 Review Summary 
 ### Release Note

**Purpose**:
- Enhance the Composio toolset by introducing a character limit feature for action names.

**Changes**:
- **New Feature**: Added a character limit for action names to prevent exceeding specified lengths.
- **Enhancement**: Optimized initialization to support the new character limit feature.
- **Test**: Updated test suite to verify correct initialization and enforcement of character limits.
- **Style**: Improved test file path formatting for better readability.

**Impact**:
- Streamlines user interactions, ensuring action names are concise, enhancing readability, reliability, and ease of use. 

<details><summary>Original Description</summary>

This PR updates base toolset to allow limiting number of characters for action name in a schema

fixes: ENG-2917

----

> [!IMPORTANT]
> Adds character limit for action names in `ComposioToolSet`, with default set in Langchain integration and tests added.
> 
>   - **Behavior**:
>     - Adds `_action_name_char_limit` to `ComposioToolSet` in `toolset.py` to limit action name length.
>     - Truncates action names in `_process_schema()` if limit is set.
>   - **Langchain Integration**:
>     - Sets default `action_name_char_limit` to 64 in `composio_langchain/toolset.py`.
>   - **Tests**:
>     - Adds `test_action_name_char_limit()` in `test_toolset.py` to verify action name truncation.
>     - Updates `TestSubclassInit` to include action name limit tests.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for 3eb8a683d2de0697cb7b275470a6c08ef0321eb7. It will automatically update as commits are pushed.</sup>

</details>

### CODE REVIEW

File: python/composio/tools/toolset.py [hunk 1]
```diff
@@ -135,12 +135,14 @@ class ComposioToolSet(WithLogger):  # pylint: disable=too-many-public-methods
 
     _runtime: str = "composio"
     _description_char_limit: int = 1024
+    _action_name_char_limit: t.Optional[int] = None
```
Comment (shreysingla11) [inline python/composio/tools/toolset.py:138]: Consider adding a docstring for the `_action_name_char_limit` class variable to explain its purpose and default behavior. For example:
```python
# Maximum length for action names. None means no limit is applied.
_action_name_char_limit: t.Optional[int] = None
```

File: python/composio/tools/toolset.py [hunk 2]
```diff
@@ -153,6 +155,7 @@ def __init_subclass__(
                 f"description_char_limit is not set on {cls.__name__}, using 1024 as default"
             )
         cls._description_char_limit = description_char_limit or 1024
+        cls._action_name_char_limit = action_name_char_limit
```
Comment (shreysingla11) [inline python/composio/tools/toolset.py:158]: Consider adding a warning message when `action_name_char_limit` is not set, similar to how we handle `description_char_limit`. This would maintain consistency in the codebase:

```python
if action_name_char_limit is None:
    warnings.warn(
        f"action_name_char_limit is not set on {cls.__name__}, using no limit as default"
    )
```

File: python/tests/test_tools/test_toolset.py [hunk 3]
```diff
@@ -328,3 +329,64 @@ def test_execute_action_param_serialization() -> None:
...
+            pass
+
+        (schema,) = SomeToolsetExtention().get_action_schemas(
+            actions=[
+                Action.FILETOOL_GIT_CLONE,
+            ]
+        )
+        assert len(t.cast(str, schema.description)) == char_limit
+
+    def test_action_name_char_limit(self) -> None:
```
Comment (shreysingla11) [inline python/tests/test_tools/test_toolset.py:374]: Consider adding test cases for edge cases:
1. Test with limit=0 (should preserve empty string)
2. Test with limit=1 (minimal valid length)
3. Test with negative limit (should raise ValueError)
4. Test with very large limit (should preserve full name)

This would help ensure robust handling of all possible input values.

**Comment by shreysingla11** [comment]
### Code Review Summary

The PR implements action name character limits in a clean and maintainable way. The implementation is solid with good test coverage.

#### Strengths:
✅ Clean implementation following existing patterns
✅ Proper type hints and documentation
✅ Maintains backward compatibility
✅ Good test coverage for main functionality
✅ Configurable per toolset implementation

#### Suggestions for Improvement:
1. Add docstring for the new `_action_name_char_limit` class variable
2. Add warning message when `action_name_char_limit` is not set (for consistency)
3. Add test cases for edge cases (0, 1, negative values, very large values)

#### Overall Assessment:
The code is well-structured and implements the feature correctly. The changes are focused and maintain backward compatibility. After addressing the minor suggestions above, this PR would be ready to merge.

Code Quality Rating: 8/10 (Very Good)
- Clean implementation: ⭐⭐⭐
- Test coverage: ⭐⭐
- Documentation: ⭐⭐
- Maintainability: ⭐

File: python/tests/test_tools/test_toolset.py [hunk 4]
```diff
@@ -328,3 +329,64 @@ def test_execute_action_param_serialization() -> None:
...
+
+    def test_description_char_limit(self) -> None:
+
+        char_limit = 512
+        (schema,) = ComposioToolSet().get_action_schemas(
+            actions=[
+                Action.FILETOOL_GIT_CLONE,
+            ]
+        )
+        assert len(t.cast(str, schema.description)) > char_limit
```
Comment (tushar-composio) [inline python/tests/test_tools/test_toolset.py:362]: why is cast needed?

Comment (tushar-composio) [reply python/tests/test_tools/test_toolset.py:362]: Data model fields should not be optional. I'll make a card for this.
